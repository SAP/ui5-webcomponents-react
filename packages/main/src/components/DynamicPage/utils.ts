export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
