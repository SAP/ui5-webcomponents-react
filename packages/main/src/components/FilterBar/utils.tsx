import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
