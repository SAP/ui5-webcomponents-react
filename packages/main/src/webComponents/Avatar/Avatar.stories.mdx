import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs';
import { Avatar } from './index';
import { AvatarColorScheme } from '../../enums/AvatarColorScheme';
import '@ui5/webcomponents-icons/dist/employee.js';
import { AvatarShape } from '../../enums/AvatarShape';
import { AvatarSize } from '../../enums/AvatarSize';
import { DocsHeader } from '@docs/DocsHeader';
import { Footer } from '@docs/Footer';

<Meta
  title="Data Display / Avatar"
  component={Avatar}
  args={{
    colorScheme: AvatarColorScheme.Accent6,
    icon: 'employee',
    shape: AvatarShape.Circle,
    size: AvatarSize.S
  }}
  argTypes={{
    children: { control: { disable: true } }
  }}
/>

<DocsHeader since="0.9.0" />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Avatar {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />

<Footer />
