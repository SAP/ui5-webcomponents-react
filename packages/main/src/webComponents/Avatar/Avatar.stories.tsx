import type { Meta, StoryObj } from '@storybook/react';
import AvatarColorScheme from '@ui5/webcomponents/dist/types/AvatarColorScheme.js';
import AvatarShape from '@ui5/webcomponents/dist/types/AvatarShape.js';
import AvatarSize from '@ui5/webcomponents/dist/types/AvatarSize.js';
import { Avatar } from './index';

const meta = {
  title: 'Data Display / Avatar',
  component: Avatar,
  argTypes: {
    children: { control: { disable: true } }
  },
  args: {
    colorScheme: AvatarColorScheme.Accent6,
    icon: 'employee',
    shape: AvatarShape.Circle,
    size: AvatarSize.S
  },
  tags: ['package:@ui5/webcomponents']
} satisfies Meta<typeof Avatar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
