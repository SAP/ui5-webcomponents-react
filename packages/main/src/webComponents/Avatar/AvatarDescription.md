## Keyboard Handling

*   \[SPACE, ENTER, RETURN\] - Fires the `click` event if the `interactive` property is set to true.
*   \[SHIFT\] - If \[SPACE\] is pressed, pressing \[SHIFT\] releases the component without triggering the click event.