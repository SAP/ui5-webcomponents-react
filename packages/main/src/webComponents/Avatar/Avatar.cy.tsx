import { cypressPassThroughTestsFactory } from '@/cypress/support/utils';
import { Avatar } from './index';

describe('Avatar', () => {
  it('Basic Test (generated)', () => {
    cy.mount(<Avatar />);
    cy.get('[ui5-avatar]').should('exist');
  });

  cypressPassThroughTestsFactory(Avatar);
});
