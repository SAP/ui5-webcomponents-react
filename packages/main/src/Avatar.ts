import { deprecationNotice } from '@ui5/webcomponents-react-base';
import type { AvatarDomRef, AvatarPropTypes } from './webComponents/Avatar';
import { Avatar } from './webComponents/Avatar';

deprecationNotice(
  'Avatar',
  `importing from '@ui5/webcomponents-react/dist/Avatar' is deprecated. Please use "import { Avatar } from '@ui5/webcomponents-react';" instead.`
);

export { Avatar };

export type { AvatarPropTypes, AvatarDomRef };
