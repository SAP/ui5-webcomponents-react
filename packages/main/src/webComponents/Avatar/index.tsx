import { AvatarColorScheme } from '@ui5/webcomponents-react/dist/AvatarColorScheme';
import { AvatarShape } from '@ui5/webcomponents-react/dist/AvatarShape';
import { AvatarSize } from '@ui5/webcomponents-react/dist/AvatarSize';
import { withWebComponent } from '@ui5/webcomponents-react/dist/withWebComponent';
import { CommonProps } from '@ui5/webcomponents-react/interfaces/CommonProps';
import { ReactNode } from 'react';

import '@ui5/webcomponents/dist/Avatar.js';

export interface AvatarPropTypes extends CommonProps {
  /**
   * Defines the text alternative of the component. If not provided a default text alternative will be set, if present.
   */
  accessibleName?: string;
  /**
   * Defines the background color of the desired image.
   *
   * Available options are:
   *
   * *   `Accent1`
   * *   `Accent2`
   * *   `Accent3`
   * *   `Accent4`
   * *   `Accent5`
   * *   `Accent6`
   * *   `Accent7`
   * *   `Accent8`
   * *   `Accent9`
   * *   `Accent10`
   * *   `Placeholder`
   */
  colorScheme?: AvatarColorScheme | keyof typeof AvatarColorScheme;
  /**
   * Defines the name of the UI5 Icon, that would be displayed.
   * **Note:** If `image` slot is provided, the property would be ignored.
   * **Note:** You should import the desired icon first, then use its name as "icon".
   *
   * import "@ui5/webcomponents-icons/dist/{icon\_name}.js"
   *
   * `<Avatar icon="employee">`
   *
   * See all the available icons in the <ui5-link target="_blank" href="https://openui5.hana.ondemand.com/test-resources/sap/m/demokit/iconExplorer/webapp/index.html" class="api-table-content-cell-link">Icon Explorer</ui5-link>.
   */
  icon?: string;
  /**
   * Defines the displayed initials.
   * Up to two Latin letters can be displayed as initials.
   */
  initials?: string;
  /**
   * Defines if the avatar is interactive (focusable and pressable)
   */
  interactive?: boolean;
  /**
   * Defines the shape of the component.
   *
   * Available options are:
   *
   * *   `Circle`
   * *   `Square`
   */
  shape?: AvatarShape | keyof typeof AvatarShape;
  /**
   * Defines predefined size of the component.
   *
   * Available options are:
   *
   * *   `XS`
   * *   `S`
   * *   `M`
   * *   `L`
   * *   `XL`
   */
  size?: AvatarSize | keyof typeof AvatarSize;
  /**
   * Receives the desired `<img>` tag
   *
   * **Note:** If you experience flickering of the provided image, you can hide the component until it is being defined with the following CSS:
   *
   * `Avatar:not(:defined) {  visibility: hidden;   }   `
   */
  children?: ReactNode;
}

/**
 * An image-like control that has different display options for representing images and icons in different shapes and sizes, depending on the use case. The shape can be circular or square. There are several predefined sizes, as well as an option to set a custom size.
 *
 * <ui5-link href="https://sap.github.io/ui5-webcomponents/playground/components/Avatar" target="_blank">UI5 Web Components Playground</ui5-link>
 */
const Avatar = withWebComponent<AvatarPropTypes>(
  'ui5-avatar',
  ['accessibleName', 'colorScheme', 'icon', 'initials', 'shape', 'size'],
  ['interactive'],
  [],
  []
);

Avatar.displayName = 'Avatar';

export { Avatar };
