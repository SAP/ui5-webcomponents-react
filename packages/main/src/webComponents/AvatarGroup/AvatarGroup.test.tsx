import { render } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { AvatarGroup } from './index';
import { Avatar } from '../Avatar';
import React from 'react';

describe('AvatarGroup', () => {
  test('Basic Test (generated)', () => {
    const { asFragment } = render(
      <AvatarGroup>
        <Avatar initials="UI5" />
      </AvatarGroup>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  createCustomPropsTest(AvatarGroup, { children: <Avatar initials="UI5" /> });
});
