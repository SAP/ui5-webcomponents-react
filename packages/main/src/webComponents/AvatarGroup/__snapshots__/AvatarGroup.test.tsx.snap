// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarGroup Basic Test (generated) 1`] = `
<DocumentFragment>
  <ui5-avatar-group
    type="Group"
    ui5-avatar-group=""
  />
</DocumentFragment>
`;
